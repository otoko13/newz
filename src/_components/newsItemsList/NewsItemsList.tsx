import React from 'react';
import { Stack } from 'office-ui-fabric-react/lib/Stack';
import { INewsItem } from '../../NewzApp';
import './newsItemsList.scss';
import ItemCard from '../itemCard/ItemCard';

export interface INewsItemsListProps {
    newsItems: INewsItem[];
    latestItemIds: number[];
    onNewsItemSelected: (newsItem: INewsItem) => void;
}

const NewsItemsList = (props: INewsItemsListProps) => {
    return (
        <Stack verticalAlign='start' className='NewsItemsList'>
            {
                props.newsItems.map(item => 
                    <ItemCard 
                        key={item.id} 
                        newsItem={item}
                        onItemSelected={() => props.onNewsItemSelected(item)}
                        isLatest={props.latestItemIds.includes(item.id)} />
                )
            }
        </Stack>
    );
}

export default NewsItemsList;