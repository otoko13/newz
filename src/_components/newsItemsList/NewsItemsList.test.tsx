import { render } from '@testing-library/react';
import * as React from 'react';
import NewsItemsList, { INewsItemsListProps } from './NewsItemsList';

let props: INewsItemsListProps;

beforeEach(() => {
    props = {
        latestItemIds: [],
        newsItems: [
            { id: 2, descendants: 0, by: 'Maurice', kids: [], time: 1581442689, title: 'Another comment', text: 'I believe', type: 'story' },
            { id: 3, descendants: 0, by: 'Phillip', kids: [], time: 1581442689, title: 'Yet another comment', text: 'I don\'t', type: 'story' },
            { id: 4, descendants: 0, by: 'Kevin', kids: [], time: 1581442689, title: 'One more comment', text: 'Well who the hell are you', type: 'story' },
        ],
        onLoadOlderNewsClicked: jest.fn(),
        onNewsItemSelected: jest.fn(),
        visitedItems: [],
    }
});

describe('snapshot', () => {
    it('should look sensible', () => {
        const { container } = render(<NewsItemsList {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});