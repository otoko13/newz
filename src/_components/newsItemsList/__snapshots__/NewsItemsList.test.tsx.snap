// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should look sensible 1`] = `
<div
  class="NewsItemsList"
  style="height: 345px;"
>
  <div
    class="ItemCard   "
    style="top: 0px; margin-bottom: 20px; height: 95px;"
  >
    <div
      class="ms-Stack css-36"
    >
      <div
        class="ms-StackItem css-37"
      >
        <div
          class="ms-Stack css-38"
        >
          <div
            class="ms-StackItem css-37"
          >
            <div
              class="title"
            >
              Another comment
            </div>
          </div>
          <div
            class="ms-StackItem icons css-37"
          >
            <i
              aria-hidden="true"
              class="title title-icon root-33"
              data-icon-name="Articles"
            />
          </div>
        </div>
      </div>
      <div
        class="ms-StackItem css-37"
      >
        <div
          class="ms-Stack details css-39"
        >
          <div
            class="ms-StackItem date css-37"
          >
            17:38, Tuesday February 11th 2020
          </div>
          <div
            class="ms-StackItem author css-37"
          >
            Maurice
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ItemCard   "
    style="top: 115px; margin-bottom: 20px; height: 95px;"
  >
    <div
      class="ms-Stack css-36"
    >
      <div
        class="ms-StackItem css-37"
      >
        <div
          class="ms-Stack css-38"
        >
          <div
            class="ms-StackItem css-37"
          >
            <div
              class="title"
            >
              Yet another comment
            </div>
          </div>
          <div
            class="ms-StackItem icons css-37"
          >
            <i
              aria-hidden="true"
              class="title title-icon root-33"
              data-icon-name="Articles"
            />
          </div>
        </div>
      </div>
      <div
        class="ms-StackItem css-37"
      >
        <div
          class="ms-Stack details css-39"
        >
          <div
            class="ms-StackItem date css-37"
          >
            17:38, Tuesday February 11th 2020
          </div>
          <div
            class="ms-StackItem author css-37"
          >
            Phillip
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ItemCard   "
    style="top: 230px; margin-bottom: 20px; height: 95px;"
  >
    <div
      class="ms-Stack css-36"
    >
      <div
        class="ms-StackItem css-37"
      >
        <div
          class="ms-Stack css-38"
        >
          <div
            class="ms-StackItem css-37"
          >
            <div
              class="title"
            >
              One more comment
            </div>
          </div>
          <div
            class="ms-StackItem icons css-37"
          >
            <i
              aria-hidden="true"
              class="title title-icon root-33"
              data-icon-name="Articles"
            />
          </div>
        </div>
      </div>
      <div
        class="ms-StackItem css-37"
      >
        <div
          class="ms-Stack details css-39"
        >
          <div
            class="ms-StackItem date css-37"
          >
            17:38, Tuesday February 11th 2020
          </div>
          <div
            class="ms-StackItem author css-37"
          >
            Kevin
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
